### PR TITLE
Adding batch to the mix

### PR DESCRIPTION
**Pull request by ecwillis**
See if this will work for you. Comments of course welcome.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,10 +24,10 @@ exports.register = function(plugin, options, next) {
   });
 
   agenda.on('fail', function(err, job) {
-    plugin.log(['agenda', 'complete'], {err: err, job: job.attrs});
+    plugin.log(['agenda', 'complete'], { err: err, job: job.attrs });
   });
 
-  if(options.jobs) {
+  if ( options.jobs ) {
```
Comment (jgallen23) [inline index.js]: `if (options.jobs) {`

File: index.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ exports.register = function(plugin, options, next) {
     var opts = {};
     var name, method;
 
-    if(typeof value === 'function') {
+    if ( typeof value === 'function' ) {
```
Comment (jgallen23) [inline index.js]: no space after (

File: index.js [hunk 3]
```diff
@@ -49,32 +49,31 @@ exports.register = function(plugin, options, next) {
     }
 
     agenda.define(name, opts, method.bind(plugin));
-  });
 
-  if(options.every) {
```
Comment (jgallen23) [inline index.js]: why was this removed?

File: index.js [hunk 4]
```diff
@@ -49,32 +49,31 @@ exports.register = function(plugin, options, next) {
...
-    });
-  }
+  });
 
   agenda.start();
 
+  var basePath = options.basePath || '';
+  var auth = options.auth || false;
+
+  var batch = new Batch( agenda );
```
Comment (jgallen23) [inline index.js]: no space after (

File: index.js [hunk 5]
```diff
@@ -49,32 +49,31 @@ exports.register = function(plugin, options, next) {
...
   }
 
-  next();
-};
+}
 
 exports.register.attributes = {
-  name: 'agenda'
-};
+  name: 'batch'
```
Comment (jgallen23) [inline index.js]: this should not be changed

File: json-handler.js [hunk 6]
```diff
@@ -77,6 +77,18 @@ module.exports = function(path, auth) {
           reply(job);
         }
       }
+    },
+    testThing: {
```
Comment (jgallen23) [inline json-handler.js]: can you explain this?

File: package.json [hunk 7]
```diff
@@ -24,13 +25,14 @@
   },
   "homepage": "https://github.com/firstandthird/hapi-agenda",
   "dependencies": {
-    "agenda": "^0.6.26",
+    "agenda-batch": "0.0.1",
```
Comment (jgallen23) [inline package.json]: we still need agenda, right?

File: example/index.js [hunk 8]
```diff
@@ -7,21 +7,42 @@ server.connection({
...
-    server.log(['server', 'info'], 'Hapi server started '+ server.info.uri);
-  });
+});
+
+
+server.route({
+  method: 'GET',
+  path: '/',
+  handler: function (request, reply) {
+      reply('hapi-agenda example. Visit <a href="/spawn-job">/spawn-job</a> to create a new agenda batch job.');
```
Comment (jgallen23) [inline example/index.js]: looks like extra spaces here

File: example/index.js [hunk 9]
```diff
@@ -7,21 +7,42 @@ server.connection({
...
+  }
+});
+
+
+server.route({
+  method: 'GET',
+  path: '/spawn-job',
+  handler: function (request, reply) {
+      reply('Spawning a new job.');
+      var batch = server.plugins['batch']['batch'];
```
Comment (jgallen23) [inline example/index.js]: server.plugins.agenda.batch - no need for `['`

File: index.js (hunk 3 above)
Comment (ecwillis) [inline index.js]: Mistake... Originally I was under the thinking that batch jobs are not created as recurring jobs so there would be no need.

File: json-handler.js (hunk 6 above)
Comment (ecwillis) [inline json-handler.js]: Just working out the hapi api. Removed.

File: package.json (hunk 7 above)
Comment (ecwillis) [inline package.json]: Technically yes and no. agenda-batch requires it too... but I added it back.

File: example/index.js [hunk 10]
```diff
@@ -13,15 +13,30 @@ server.register({
...
+      reply('hapi-agenda example. Visit <a href="/spawn-job">/spawn-job</a> to create a new agenda batch job.');
+  }
+});
+
+server.route({
+  method: 'GET',
+  path: '/spawn-job',
+  handler: function (request, reply) {
+      reply('Spawning a new job.');
+      var batch = server.plugins['agenda']['batch'];
```
Comment (jgallen23) [inline example/index.js]: `server.plugins.agenda.batch`
